modin/pandas/groupby.py

```diff
@@ -301,29 +302,40 @@ def aggregate(self, func=None, *args, **kwargs):
             # so we throw a different message
             raise NotImplementedError("axis other than 0 is not supported")
         if isinstance(func, dict) or func is None:
-            if func is None:
-                func = {}
-            else:
-                if any(i not in self._df.columns for i in func.keys()):
-                    from pandas.core.base import SpecificationError
 
-                    raise SpecificationError("nested renamer is not supported")
+            def _reconstruct_func(func, **kwargs):
+                relabeling_required, func, new_columns, order = reconstruct_func(
```

Comment:
@dchigarev please describe your solution.

What does the `reconstruct_func` function do?

Reply:
`groupby_dict_agg` query compilers method, assumes that we're passing our dictionary function like:
`{"col_name_to_apply": [func1, func2, ...]}`
hovewer pandas also support cases like this:
`{"new_column_name": ("col_name_to_apply", func)}`
`reconstruct_func` is internal pandas function, that can cast second case to the first + give us meta information about relabeling result.
More in [pandas doc](https://github.com/pandas-dev/pandas/blob/db08276bc116c438d3fdee492026f8223584c477/pandas/core/aggregation.py#L35-L70)